File: tests/sdk/go/forensic_lab_test.go
package sdk_test

import (
	"os"
	"testing"
	"time"
)

func TestForensicLabSetClockAndBulkSeed(t *testing.T) {
	requireReady(t)
	if os.Getenv("NOCTAXRIS_LAB_FORENSICS") != "1" {
		t.Skip("set NOCTAXRIS_LAB_FORENSICS=1 on the API process for NoctaxrisLab SetClock/BulkSeed")
	}

	fixed := time.Date(2026, 7, 20, 12, 0, 0, 0, time.UTC)
	setStatus, setBody, setParsed := signedJSONTarget(t, "noctaxris", "NoctaxrisLab.SetClock", map[string]any{
		"FixedTime": fixed.Format(time.RFC3339),
	})
	if setStatus != 200 {
		t.Fatalf("SetClock status=%d body=%s", setStatus, setBody)
	}
	if clock, _ := setParsed["ClockTime"].(string); clock != fixed.Format(time.RFC3339) {
		t.Fatalf("SetClock ClockTime=%v want %s body=%s", setParsed["ClockTime"], fixed.Format(time.RFC3339), setBody)
	}

	seedStatus, seedBody, seedParsed := signedJSONTarget(t, "noctaxris", "NoctaxrisLab.BulkSeed", map[string]any{
		"ScenarioId":       "crypto-mining",
		"IncludeGuardDuty": true,
	})
	if seedStatus != 200 {
		t.Fatalf("BulkSeed status=%d body=%s", seedStatus, seedBody)
	}
	if sid, _ := seedParsed["ScenarioId"].(string); sid != "crypto-mining" {
		t.Fatalf("BulkSeed ScenarioId=%v body=%s", seedParsed["ScenarioId"], seedBody)
	}
	ctCount, _ := seedParsed["CloudTrailEventCount"].(float64)
	if ctCount < 1 {
		t.Fatalf("BulkSeed CloudTrailEventCount=%v body=%s", seedParsed["CloudTrailEventCount"], seedBody)
	}
	gdCount, _ := seedParsed["GuardDutyFindingCount"].(float64)
	if gdCount < 1 {
		t.Fatalf("BulkSeed GuardDutyFindingCount=%v body=%s", seedParsed["GuardDutyFindingCount"], seedBody)
	}
}

File: tests/sdk/nodejs/test/forensic-lab.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import { requireReady, signedJsonTarget } from "../lib/helpers.mjs";

test("Lab forensics SetClock and BulkSeed crypto-mining", async (t) => {
  if (!(await requireReady(t))) return;
  if (process.env.NOCTAXRIS_LAB_FORENSICS !== "1") {
    t.skip(
      "set NOCTAXRIS_LAB_FORENSICS=1 on the API process for NoctaxrisLab SetClock/BulkSeed",
    );
    return;
  }

  const fixed = "2026-07-20T12:00:00Z";
  const setClock = await signedJsonTarget("noctaxris", "NoctaxrisLab.SetClock", {
    FixedTime: fixed,
  });
  assert.equal(setClock.status, 200, setClock.body);
  assert.equal(
    setClock.json?.ClockTime,
    fixed,
    `SetClock ClockTime=${setClock.json?.ClockTime} body=${setClock.body}`,
  );

  const seed = await signedJsonTarget("noctaxris", "NoctaxrisLab.BulkSeed", {
    ScenarioId: "crypto-mining",
    IncludeGuardDuty: true,
  });
  assert.equal(seed.status, 200, seed.body);
  assert.equal(seed.json?.ScenarioId, "crypto-mining", seed.body);
  assert.ok(
    (seed.json?.CloudTrailEventCount || 0) >= 1,
    `BulkSeed CloudTrailEventCount=${seed.json?.CloudTrailEventCount} body=${seed.body}`,
  );
  assert.ok(
    (seed.json?.GuardDutyFindingCount || 0) >= 1,
    `BulkSeed GuardDutyFindingCount=${seed.json?.GuardDutyFindingCount} body=${seed.body}`,
  );
});
